README.md

```diff
@@ -158,6 +158,39 @@ And here's an example of how you might document a class:
      ) { }
 ```
 
+Puppet providers often rely on different types of metaprogramming to create
+parameters and methods. Since Strings does not execute the code which is being
+documented you must provide hints on how to document your code. To document a
+parameter which is automatically created you must use the special directive
+`@!puppet.provider.param` Which may take types, the parameter name, and a
+description.
+
+Strings will automatically extract the `@doc` provider docstring and any `desc`
+parameter docstrings.
+
+```ruby
+# @!puppet.provider.param my_parameter This parameter needs to be explicitly
```

Comment:
Actually that wouldn't be correct here. The directive is for documenting things which strings doesn't pick up on automatically. Strings will document the `path` parameter, but it will not document `my_param` which we are pretending will be generated automatically by metaprogramming.
Does that make sense?